ui/src/app/cohort-review/annotation-definition-modals/annotation-definition-modals.component.tsx

```diff
@@ -0,0 +1,284 @@
+import {Component, Input} from '@angular/core';
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+import {validate} from 'validate.js';
+
+import {Button, Clickable} from 'app/components/buttons';
+import {styles as headerStyles} from 'app/components/headers';
+import {ClrIcon} from 'app/components/icons';
+import {Select, TextInput} from 'app/components/inputs';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {TooltipTrigger} from 'app/components/popups';
+import {cohortAnnotationDefinitionApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, ReactWrapperBase, summarizeErrors, withUrlParams} from 'app/utils/index';
+import {AnnotationType, CohortAnnotationDefinition} from 'generated/fetch';
+
+const styles = reactStyles({
+  editRow: {
+    display: 'flex', alignItems: 'center',
+    height: '2rem', borderBottom: '1px solid #c3c3c3'
+  },
+  defName: {
+    flex: 1, whiteSpace: 'nowrap', overflow: 'hidden', textOverflow: 'ellipsis'
+  }
+});
+
+export const AddAnnotationDefinitionModal = withUrlParams()(class extends React.Component<
+  {
+    annotationDefinitions: CohortAnnotationDefinition[],
+    onCancel: Function,
+    onCreate: Function,
+    urlParams: any
+  },
+  {name: string, annotationType: AnnotationType, enumValues: string[], saving: boolean}
+> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      annotationType: AnnotationType.STRING,
+      enumValues: [],
+      saving: false
+    };
+  }
+
+  async create() {
+    try {
+      const {onCreate, urlParams: {ns, wsid, cid}} = this.props;
+      const {name, annotationType, enumValues} = this.state;
+      this.setState({saving: true});
+      const newAd = await cohortAnnotationDefinitionApi().createCohortAnnotationDefinition(
```

Comment:
Nit: I'd call this `newAnnotationDefinition`, especially since it isn't used very often? I think its more understandable.

Also, can this be inlined below?

Reply:
It could be inlined, yes. I usually do it this way for readability, to highlight the `await`, otherwise it can get lost inside the expression. The other confounding issue here is that a longer name would go past the line limit and require additional restructuring. (FWIW I don't like line limits for that reason, but that's another discussion)